### DISCUSSION TITLE
How should I go about integrating large models (for machine learning) with GitHub?

### DISCUSSION THREAD

**Discussion by rafa-br34** [community/community#124452] [Programming Help] [answered]
### Select Topic Area

Question

### Body

Hello! So I plan on creating a repository with large models in the future, and I wonder what would be the best approach to seamlessly integrate code and large models, whether I should use a third party (huggingface for example), make a repository for code on GitHub and host the models somewhere else, or if there's any way I could use GitHub entirely.
Forgive me if a similar question has already been asked, I have done light research on the matter but I didn't find similar questions.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @rafa-br34 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by CopperEagle** [comment, marked as answer]
Hey @rafa-br34 

It depends on what you are going for :smiley:  Let me give you options.

- If you just need some AI Services (ChatGPT, Claude, ...) you can use their APIs. (like [that one](https://pypi.org/project/openai/)). No special managing of anything, as these are pretty much just function calls.
- If you need one or several models that are already on Huggingface: I recommend using classes and inheritance to abstract the different implementations away, providing only a simple interface like "load()", "chat(...)", "save()".
- If you develop your own model and then use it in your code: Put the model on Huggingface, then apply the above. Note that individual Github repositories typically do not have enough storage for big AI models. (You shouldn't really go much beyond 1 GB.) So Huggingface is probably the place for the AI model.

If you want a real code example using option 2, you can have a look at my code [here](https://github.com/CopperEagle/SmartFileLibrary/tree/main/smartfilelibrary/analyzers). It provides several different options to analyze PDF documents, using different Huggingface models for each option. A baseclass provides basic functionality and defines the interface.
